src/lib/arch/aarch64/Architecture.cc

```diff
@@ -51,12 +47,13 @@ Architecture::Architecture(kernel::Linux& kernel, YAML::Node config)
   }
   // Extract execution latency/throughput for each group
   std::vector<uint8_t> inheritanceDistance(NUM_GROUPS, UINT8_MAX);
-  for (size_t i = 0; i < config["Latencies"].size(); i++) {
-    YAML::Node port_node = config["Latencies"][i];
+  for (size_t i = 0; i < config["Latencies"].num_children(); i++) {
+    ryml::ConstNodeRef port_node = config["Latencies"][i];
     uint16_t latency = port_node["Execution-Latency"].as<uint16_t>();
     uint16_t throughput = port_node["Execution-Throughput"].as<uint16_t>();
-    for (size_t j = 0; j < port_node["Instruction-Group"].size(); j++) {
-      uint16_t group = port_node["Instruction-Group"][j].as<uint16_t>();
+    for (size_t j = 0; j < port_node["Instruction-Group-Nums"].num_children();
+         j++) {
+      uint16_t group = port_node["Instruction-Group-Nums"][j].as<uint16_t>();
```

Comment:
I may have missed this, but where does `"Instruction-Group-Nums"` come from? Added on config validation?

Reply:
It stores the string to int conversions created in `ModelConfig::postValidation()`